.github/workflows/cicd.yaml

```diff
@@ -141,6 +141,93 @@ jobs:
           check_name: Linux Test Report
           comment_title: Linux Test Report
 
+  macos-cpp:
+    runs-on: macos-latest
+    if: github.repository == '4paradigm/OpenMLDB'
+    container:
+      image: ghcr.io/4paradigm/hybridsql:latest
+    env:
+      OS: darwin
+      ARCH: x86_64
+      NPROC: 3
+      CTEST_PARALLEL_LEVEL: 3 # parallel test level for ctest (make test)
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: prepare release
+        run: |
+          VERSION="snapshot"
+          if [[ ${{ github.ref }} == "refs/tags/v"* ]]; then
+            VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+            VERSION=${VERSION#v}
+            bash steps/prepare_release.sh "$VERSION"
+          fi
+          echo "OPENMLDB_PREFIX=openmldb-$VERSION-macos" >> $GITHUB_ENV
+
+      - name: enable java on release
+        if: github.event_name == 'push'
+        run: |
+          echo "SQL_JAVASDK_ENABLE=ON" >> $GITHUB_ENV
+
+      - name: configure
+        env:
+          SQL_JAVASDK_ENABLE: ${{ env.SQL_JAVASDK_ENABLE }}
+        run: |
+          make configure CMAKE_INSTALL_PREFIX=${{ env.OPENMLDB_PREFIX }}
+
+      - name: build
+        run: |
+          make build
+          # GitHub runner disk space is limited
+          # delete thirdparty build directory($ROOT/.deps/) to save disk space
+          # make thirdpartybuild-clean
+
+      - name: run cpp
+        id: unittest
+        run: |
+          make test
+
+      - name: run sql_sdk_test
+        id: sql_sdk_test
+        run: |
+          bash steps/ut.sh sql_sdk_test 0
+
+      - name: run sql_cluster_test
+        id: sql_cluster_test
+        run: |
+          bash steps/ut.sh sql_cluster_test 0
+
+      - name: run sql_standalone_sdk_test
+        id: sql_standalone_sdk_test
+        run: |
+          bash steps/ut.sh sql_standalone_sdk_test 0
+
+      - name: upload unit test results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: linux-ut-result-cpp-${{ github.sha }}
+          path: |
+            build/**/*.xml
+            reports/*.xml
+
+      - name: install
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          make install
+
+      - name: package
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          tar czf ${{ env.OPENMLDB_PREFIX }}.tar.gz ${{ env.OPENMLDB_PREFIX }}/
+
+      - name: upload artifacts
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/upload-artifact@v2
+        with:
+          path: openmldb-*.tar.gz
+          name: release-artifacts
+
```

Comment:
release dependencies need update https://github.com/4paradigm/OpenMLDB/pull/2028/files#diff-2c9df400afaffa9270cab617792f08cc671e5f95ae192f1cf111ef4e990a4addR234

Reply:
I do not really understand, could you please give me some advice?

Reply:
So basically the release flow is `tag push` -> `trigger this cicd workflow and runs cpp and cpp-macos` -> `runs the release job below` , and finally you can find the new release in https://github.com/4paradigm/OpenMLDB/releases. 

This step uploaded the macos artifact just built here, you can also find the same step in `cpp` job. And the `release` job just download linux and macos artifacts and release them once for all

Reply:
> So basically the release flow is `tag push` -> `trigger this cicd workflow and runs cpp and cpp-macos` -> `runs the release job below` , and finally you can find the new release in https://github.com/4paradigm/OpenMLDB/releases.
> 
> This step uploaded the macos artifact just built here, you can also find the same step in `cpp` job. And the `release` job just download linux and macos artifacts and release them once for all


This flow seems is consistent with you described~, whether the `runs-on` in `release` step will affect the action of release linux and macos artifacts?

Reply:
> > So basically the release flow is `tag push` -> `trigger this cicd workflow and runs cpp and cpp-macos` -> `runs the release job below` , and finally you can find the new release in https://github.com/4paradigm/OpenMLDB/releases.
> > This step uploaded the macos artifact just built here, you can also find the same step in `cpp` job. And the `release` job just download linux and macos artifacts and release them once for all
> 
> This flow seems is consistent with you described~, whether the `runs-on` in `release` step will affect the action of release linux and macos artifacts?

https://github.com/4paradigm/OpenMLDB/blob/af0272cec5b9646c8b7eaa3c6d9e9dfccaf6f56d/.github/workflows/cicd.yaml#L144-L145

Reply:
the release job just publish the built artifacts of needed two job, so it do not matter which platform release job runs on

Reply:
Got, thanks.